em_examples/TDEMHorizontalLoopCylWidget.py

```diff
@@ -0,0 +1,775 @@
+        )
+        Ey = self.mesh.aveE2CC*self.f[src, "e", itime]
+        Jy = Utils.sdiag(self.prb.sigma) * Ey
+
+        self.Ey = Utils.mkvc(
+            self.mirrorArray(Ey[self.activeCC], direction="y")
+        )
+        self.Jy = Utils.mkvc(
+            self.mirrorArray(Jy[self.activeCC], direction="y")
+        )
+        self.Bx = Utils.mkvc(
+            self.mirrorArray(Pfx*self.f[src, "b", itime], direction="x")
+        )
+        self.Bz = Utils.mkvc(
+            self.mirrorArray(Pfz*self.f[src, "b", itime], direction="z")
+        )
+        self.dBxdt = Utils.mkvc(self.mirrorArray(
+            -Pfx*self.mesh.edgeCurl*self.f[src, "e", itime], direction="x")
+        )
+        self.dBzdt = Utils.mkvc(self.mirrorArray(
+            -Pfz*self.mesh.edgeCurl*self.f[src, "e", itime], direction="z")
+        )
+
+    def getData(self):
+        src = self.srcList[0]
+        Pfx = self.mesh.getInterpolationMat(self.rxLoc, locType="Fx")
+        Pfz = self.mesh.getInterpolationMat(self.rxLoc, locType="Fz")
+        Pey = self.mesh.getInterpolationMat(self.rxLoc, locType="Ey")
+
+        self.Ey = (Pey*self.f[src, "e", :]).flatten()
+        self.Bx = (Pfx*self.f[src, "b", :]).flatten()
+        self.Bz = (Pfz*self.f[src, "b", :]).flatten()
+        self.dBxdt = (-Pfx*self.mesh.edgeCurl*self.f[src, "e", :]).flatten()
+        self.dBzdt = (-Pfz*self.mesh.edgeCurl*self.f[src, "e", :]).flatten()
+
+    def plotField(
+        self, Field='B', view="vec", scale="linear", itime=0, Geometry=True, Scenario=None
+    ):
+        # Printout for null cases
+        if (Field == "B") & (view == "y"):
+            print("Think about the problem geometry. There is NO By in this case.")
+        elif (Field == "dBdt") & (view == "y"):
+            print("Think about the problem geometry. There is NO dBy/dt in this case.")
+        elif (Field == "E") & (view == "x") | (Field == "E") & (view == "z"):
+            print("Think about the problem geometry. There is NO Ex or Ez in this case. Only Ey.")
+        elif (Field == "J") & (view == "x") | (Field == "J") & (view == "z"):
+            print("Think about the problem geometry. There is NO Jx or Jz in this case. Only Jy.")
+        elif (Field == "E") & (view == "vec"):
+            print("Think about the problem geometry. E only has components along y. Vector plot not possible")
+        elif (Field == "J") & (view == "vec"):
+            print("Think about the problem geometry. J only has components along y. Vector plot not possible")
+        elif Field == "Model":
+            fig = plt.figure(figsize=(7, 6))
+            ax = plt.subplot(111)
+            if Scenario == 'Sphere':
+                model2D, mapping2D = self.getCoreModel('Sphere')
+            elif Scenario == 'Layer':
+                model2D, mapping2D = self.getCoreModel('Layer')
+
+            out = self.mesh2D.plotImage(np.log10(mapping2D * model2D), ax=ax)
+            cb = plt.colorbar(
```

Comment:
same comment as above for the colorbar and getting it to show up for a log scale